### PR TITLE
Support `scope` when configuring OAuth details

### LINKED ISSUE #4102: Add OAuth scope field to deploy modal
_Related to https://github.com/camunda/product-hub/issues/2089._

### Problem you would like to solve

As a user, I want to be able to deploy to Zeebe clusters that use OAuth2 as authentication mechanism with a provider that requires a `scope` in the token request, e.g. Microsoft Entra ID.

### Proposed solution

Add a field "Scope" (or better "OAuth Scope" and rename "Audience" to "OAuth Audience") to the deploy modal which gets picked up by the Zeebe client for acquiring a token.

### Alternatives considered

None

### Additional context

- Web Modeler is also missing the field, see https://github.com/camunda/web-modeler/issues/8027
- https://github.com/camunda-community-hub/zeebe-client-node-js has to provide support for passing a `scope` within the token request, see https://github.com/camunda-community-hub/camunda-8-js-sdk/issues/21
- reported via [Slack](https://camunda.slack.com/archives/C06112NS4GJ/p1707169292640159?thread_ts=1706905378.492159&cid=C06112NS4GJ)

----

Depends on https://github.com/camunda-community-hub/zeebe-client-node-js/pull/363

### PR DESCRIPTION
**Pull request by nikku**
### TLDR

* `scope` is an (optional) new field inside the deploy config modal
* `audience` and scope renamed (UI wise) to `OAuth ***`

---

image

Closes #4102

### CODE REVIEW

**Review by barmac — APPROVED** [review]
I tested this locally and the scope is passed correctly.
